src/components/accounts/RecoverWithLink.js

```diff
@@ -146,7 +146,8 @@ class RecoverWithLink extends Component {
                     this.setState({ successView: false });
                 } else {
                     this.props.refreshAccount();
-                    this.props.history.push('/profile');
+                    // TODO: Should this use Redux action to navigate?
+                    this.props.history.push(`/profile/${this.state.accountId}`);
```

Comment:
I'm not sure why we'd use Redux here? Can you please provide more info.

Reply:
@Patrick1904 maybe it already works fine, but I think it might be needed so that redux dev tools don't conflict with history changes (but instead record them as actions).

In any case it's kinda trivial:
https://github.com/supasate/connected-react-router/blob/master/FAQ.md#with-storedispatch